src/ethereum_clis/clis/execution_specs.py

```diff
@@ -76,12 +78,22 @@ def start_server(self):
             ],
         )
         start = time.time()
+        wait_time = 0.1
+        while not self.server_file_path.exists():
+            time.sleep(wait_time)
+            wait_time *= 2
         while True:
-            if self.server_file_path.exists():
+            try:
+                # This will probably 404, but we are only checking whether connections are
+                # accepted.
+                Session().get(self.server_url + "heartbeat/")
                 break
-            if time.time() - start > DAEMON_STARTUP_TIMEOUT_SECONDS:
-                raise Exception("Failed starting ethereum-spec-evm subprocess")
-            time.sleep(0)  # yield to other processes
+            except ConnectionError:
```

Comment:
I have a docker setup to make the issue easier to reproduce (using `--cpuset-cpus` to reduce the amount of CPUs available to a number similar to github CI) and I think this is close to being the correct solution, and the only issue is that it seems like this is not the right exception to catch, because changing this line to:
```python
except Exception:
```
seem to fix the issue.

I'm going to try to find which is the correct exception we should be using here and I'm thinking that we could move this check to `_evaluate_server` and simply retry there in case of error, in order to not slow down this portion of the code unless there's an issue.